pkg/asset/installconfig/aws/validation.go

```diff
@@ -576,6 +577,10 @@ func ValidateForProvisioning(client API, ic *types.InstallConfig, metadata *Meta
 		return nil
 	}
 
+	if ic.AWS.UserProvisionedDNS == dns.UserProvisionedDNSEnabled {
+		return nil
```

Comment:
Nit: maybe a debug log here that we're skipping validation because of custom DNS?